config/fi_provider.m4

```diff
@@ -51,17 +51,24 @@ dnl
 dnl HAVE_$1_DL: same value as $1_dl
 dnl
 AC_DEFUN([FI_PROVIDER_SETUP],[
-	AC_MSG_NOTICE([*** Configuring $1 provider])
-	AC_ARG_ENABLE([$1],
-	      [AS_HELP_STRING([--enable-$1@<:@=yes|no|auto|dl|PATH@:>@],
-			      [Enable $1 provider @<:@default=auto@:>@
-				(yes: enable $1 provider; no: disable $1 provider;
-				auto: enable $1 provider if possible;
-				dl: enable $1 provider and build as a loadable library;
-				PATH: enable $1 provider and use $1 installed under PATH)])
-	      ],
-	      [],
-	      [enable_$1=auto])
+
+dnl
+dnl Check if --enable-direct is being used.  If yes, exclude all other providers
+dnl
+	AS_IF([test x$enable_direct != x"no" && test x$enable_direct != x"$1"],
```

Comment:
`x$enable_direct` should be in quotes in both instances.

I know this is nit-picky, but it protects from a bozo user from doing something like `./configure "--enable-direct=foo bar baz"` (which is stupid / nonsensical, but *someone* will do it, even if by accident...).